enterprise/api_client/enterprise_catalog.py

```diff
@@ -162,11 +164,12 @@ def get_content_metadata(self, enterprise_customer, enterprise_catalogs=None):
                 for item in utils.traverse_pagination(response, endpoint):
                     content_id = utils.get_content_metadata_item_id(item)
                     content_metadata[content_id] = item
-            except (SlumberBaseException, ConnectionError, Timeout) as exc:
-                LOGGER.exception(
-                    'Failed to get content metadata for Catalog [%s] in enterprise-catalog due to: [%s]',
-                    catalog_uuid, str(exc)
-                )
+            except SlumberBaseException as exc:
+               LOGGER.exception(
+                   'Failed to get content metadata for Catalog [%s] in enterprise-catalog due to: [%s]',
+                   catalog_uuid, str(exc)
+               )
+               raise
```

Comment:
I only updated get_content_metadata for now to just fix the integrated_channels issue. I don't think other calls returning empty dicts/lists/etc. is an issue atm. It's just integrated channels has to handle someone's catalog being removed/changed and thus equates empty items to items needing deletion thus the raise here.

This does end up causing a massive waterfalling exception traceback, but I really don't want to return something here.

Reply:
I think the alternative would be to raise, catch, raise, etc. up the chain and only log the exception at the top level, but that's also the most useless part of the traceback.

Reply:
In general my thinking is that raising is almost always better than dropping exceptions. I like it, and if your local testing shows it works for the issue you tested, I am good with it. Interesting no test changes were needed..

Reply:
Yeah that also threw me off. Though I feel like if someone wrote a test like "exception raised from catalog and returns empty" they would have fixed it to not work like that lol.

Reply:
I think I'm with Binod on this one, I like raising and catching.

Reply:
yeah there was always going to be raising here. It's just about whether we raise and catch up the chain to prevent printing every waterfall exception or not. I don't think it's a _huge_ deal though.

Reply:
As far as testing I may add a test that actually checks that an exception is raised. I think that's probably what is missing here (or perhaps there is a test that verifies the log line from the exception and nothing else and since I kept the log line it doesn't affect the test - either way I'm gonna fix that)